app/commands/generate_missing_mappings_command.rb

```diff
@@ -1,185 +1,236 @@
-  MAX_RETRIES = 3
-  QDRANT_PORT = 6333
+  UNSTABLE = "unstable"
   EMBEDDING_MODEL = "text-embedding-3-small"
   MAPPING_GRADER_GPT_MODEL = "gpt-4"
 
   usage do
     no_command
   end
 
+  environment :openai_api_key do
+    desc "OpenAI API key for mappings generation"
+    required
+  end
+
+  environment :openai_url do
+    desc "OpenAI API URL for mappings generation"
+    default "https://openai-proxy.shopify.ai/v1" # TODO: we should probably not have this in this repo
+  end
+
+  environment :qdrant_url do
+    desc "Qdrant API URL for embeddings search"
+    default "http://localhost:6333"
+  end
+
+  option :shopify_version do
+    desc "Target shopify taxonomy version"
+    short "-V"
+    long "--version string"
+    default UNSTABLE
+  end
+
+  option :retries do
+    desc "Number of retries for OpenAI API"
+    short "-r"
+    long "--retries integer"
+    default 3
+    convert :to_i
+  end
+
   def execute
+    setup_options
     frame("Generating missing mappings") do
+      logger.headline("Target Shopify version: #{params[:version]}")
+      logger.headline("OpenAI url: #{params[:openai_url]}")
+      logger.headline("Qdrant url: #{params[:qdrant_url]}")
+
       find_unmapped_categories
-      return if @unmapped_category_groups.empty?
+      return if @unmapped_categories.empty?
 
-      generate_missing_mappings_for_groups
+      generate_missing_mappings
     end
   end
 
   private
 
+  def setup_options
+    params[:shopify_version] ||= sys.read_file("VERSION").strip
+  end
+
   def find_unmapped_categories
-    spinner("Searching Shopify categories that lack mappings") do
-      @unmapped_category_groups = []
-      all_shopify_category_ids = Set.new(Category.all.pluck(:id))
-      latest_shopify_version = "shopify/#{sys.read_file("VERSION").strip}"
-      MappingRule.where(
-        input_version: latest_shopify_version,
-      ).group_by(&:output_version).each do |output_version, mappings|
-        shopify_category_ids_from_mappings_input = Set.new(
-          mappings.map do |mapping|
-            mapping.input.product_category_id.split("/").last
-          end,
-        )
-        unmapped_category_ids = all_shopify_category_ids - shopify_category_ids_from_mappings_input
-        category_ids_full_names = unmapped_category_ids.sort.map do |id|
-          category_full_name = Category.find(id)&.full_name
-          [id, category_full_name] if category_full_name
-        end.compact.to_h
-        next if category_ids_full_names.empty?
-
-        @unmapped_category_groups << {
-          input_taxonomy: mappings.first.input_version,
-          output_taxonomy: output_version,
-          category_ids_full_names: category_ids_full_names,
-        }
+    spinner("Finding Shopify categories that are unmapped") do
+      all_shopify_ids = Set.new(Category.all.pluck(:id))
+      mappings_by_output = MappingRule
+        .where(input_version: "shopify/#{params[:shopify_version]}")
```

Comment:
Is it possible to move version-related changes out of the tooling scope? This will cause the tool to break.

Reply:
For now though, have remove the file parsing, which will ensure this stays as just `unstable` until we're ready for broader changes